server/controllers/personal/personalController.ts

```diff
@@ -282,4 +287,100 @@ export default class PersonalController {
+        const errors = req.flash('errors')
+
+        const prisonerBannerName = formatName(firstName, null, lastName, { style: NameFormatStyle.lastCommaFirst })
+
+        const characteristics = await this.personalPageService.getPhysicalCharacteristics(clientToken, fieldName)
+        const prisonPerson = await this.personalPageService.getPrisonPerson(clientToken, prisonerNumber, true)
+        // TODO remove when API returns value
+        if (prisonPerson) {
+          prisonPerson.physicalCharacteristics = {
+            hair: { code: '', description: '' },
+            facialHair: { code: '', description: '' },
+            faceShape: { code: '', description: '' },
+            build: { code: '', description: '' },
+          }
+        }
+
+        const fieldValue =
+          fieldValueFlash.length > 0 ? fieldValueFlash[0] : prisonPerson?.physicalCharacteristics[fieldName]
+
+        await this.auditService.sendPageView({
+          user: res.locals.user,
+          prisonerNumber: prisonerData.prisonerNumber,
+          prisonId: prisonerData.prisonId,
+          correlationId: req.id,
+          page: auditPage,
+        })
+
+        res.render('pages/edit/radios', {
+          pageTitle,
+          prisonerNumber,
+          breadcrumbPrisonerName: prisonerBannerName,
+          errors: hasLength(errors) ? errors : [],
+          hintText,
+          options: objectToSelectOptions(characteristics, 'code', 'description', fieldValue),
+          miniBannerData: {
+            prisonerName: prisonerBannerName,
+            prisonerNumber,
+            cellLocation: formatLocation(cellLocation),
+          },
+        })
+      },
+
+      submit: async (req: Request, res: Response, next: NextFunction) => {
+        const { pageTitle, fieldName, url } = fieldData
+        const { prisonerNumber } = req.params
+        const { clientToken } = req.middleware
+        const { radioField } = req.body
+        const errors: HmppsError[] = []
+
+        try {
+          await this.personalPageService.updatePhysicalCharacteristics(clientToken, prisonerNumber, {
+            [fieldName]: radioField,
+          })
+          req.flash('flashMessage', { text: `${pageTitle} updated`, type: FlashMessageType.success, fieldName })
+
+          this.auditService
+            .sendPostSuccess({
+              user: res.locals.user,
+              prisonerNumber,
+              correlationId: req.id,
+              action: PostAction.EditPhysicalCharacteristics,
+              details: { pageTitle, fieldName, radioField, url },
+            })
+            .catch(error => logger.error(error))
+
+          return res.redirect(`/prisoner/${prisonerNumber}/personal#appearance`)
+        } catch (e) {
+          errors.push({ text: 'There was an error please try again' })
+        }
+
+        req.flash('fieldValue', radioField)
+        req.flash('errors', errors)
+        return res.redirect(`/prisoner/${prisonerNumber}/personal/edit/${url}`)
```

Comment:
Why not have this in the catch block then we don't need to instantiate an empty array and mutate it?